src/main/java/mekanism/common/integration/OreDictManager.java

```diff
@@ -336,15 +402,48 @@ public static void addStandardOredictMetal(String suffix) {
                 RecipeHandler.addEnrichmentChamberRecipe(StackUtils.size(ore, 1), StackUtils.size(dusts.get(0), 2));
             }
 
-            for (ItemStack ore : OreDictionary.getOres("ingot" + suffix)) {
-                RecipeHandler.addCrusherRecipe(StackUtils.size(ore, 1), StackUtils.size(dusts.get(0), 1));
+            for (ItemStack ingot : OreDictionary.getOres("ingot" + suffix)) {
+                RecipeHandler.addCrusherRecipe(StackUtils.size(ingot, 1), StackUtils.size(dusts.get(0), 1));
+            }
+        }
+
+        if (ores.size() > 0) {
+            for (ItemStack dust : dusts) {
+                RecipeHandler.addCombinerRecipe(StackUtils.size(dust, 8), new ItemStack(Blocks.COBBLESTONE),
+                      StackUtils.size(ores.get(0), 1));    
+            }
+        }
+    }
+    
+    public static void addStandardOredictGem(String suffix) {
+        NonNullList<ItemStack> gems = OreDictionary.getOres("gem" + suffix);
+        NonNullList<ItemStack> dusts = OreDictionary.getOres("dust" + suffix);
+        NonNullList<ItemStack> ores = OreDictionary.getOres("ore" + suffix);
+        if (gems.size() > 0) {
+            for (ItemStack ore : ores) {
+                RecipeHandler.addEnrichmentChamberRecipe(StackUtils.size(ore, 1), StackUtils.size(gems.get(0), 2));
+            }
+        }
+
+        if (dusts.size() > 0) {
+            for (ItemStack gem : gems) {
+                RecipeHandler.addEnrichmentChamberRecipe(StackUtils.size(dusts.get(0), 1), StackUtils.size(gem, 1));
+                RecipeHandler.addCrusherRecipe(StackUtils.size(gem, 1), StackUtils.size(dusts.get(0), 1));
             }
         }
 
         if (ores.size() > 0) {
-            for (ItemStack ore : dusts) {
-                RecipeHandler.addCombinerRecipe(StackUtils.size(ore, 8), new ItemStack(Blocks.COBBLESTONE),
-                      StackUtils.size(ores.get(0), 1));
+            if (dusts.size() > 0) {
```

Comment:
I am not necessarily against this, but I am unsure if it is a good idea to be changing the combiner recipes from using the gem directly to using the dust. For one thing it will break people's CraftTweaker scripts if they are trying to remove the old recipe and now that recipe no longer exists (which would fail gracefully with a warning), except the dust to ore recipe would not get removed and they may not understand why the script no longer works. Not to mention I am unsure how to feel about the recipe changing based on the mods installed, (such as if there is emerald dust or not then that may be a bit annoying for pack makers, however that is less of a concern than breaking existing things).

It depends on what @thiakil thinks, but I think the best solution here may be to leave a TODO saying to switch to dust where possible for gems in Mekanism v10. Given that is a large enough version change that we can make breaking changes.

Reply:
@pupnewfster "I am unsure if it is a good idea to be changing the combiner recipes from using the gem directly to using the dust" this is only a change for the Materials without a dust added by Mekanism(diamond already uses the dust) and i don't like to use a different version of that Method for Diamond, so that is the detection mainly for handling diamonds correctly.(and i was thinking it's a good idea to use dust for everything if they are allready used for diamonds)

Reply:
By the Way the Only Recipe that is different than in the last version is that one for Emerald Ore.
All other Recipes that Method adds are new.

Reply:
Ah I could have sworn it was 3 diamonds in the combiner to make the ore not three dust. Seems I was wrong then. Looks fine overall then

Reply:
Should i add the remaining BoP gems to the list of supported gems?

Reply:
Might as well

Reply:
I Implemented this.